basis/concurrency/distributed/distributed.factor

```diff
@@ -37,22 +59,46 @@ SYMBOL: local-node
 : start-node ( addrspec -- )
     <node-server> start-server local-node set-global ;
 
-TUPLE: remote-thread node id ;
+TUPLE: remote-thread node id connection ;
+
+: <remote-thread> ( node id -- remote-thread )
+    f remote-thread boa ;
+
+TUPLE: connection remote stream local ;
 
-C: <remote-thread> remote-thread
+C: <connection> connection
+
+: connect ( remote-thread -- )
+    dup node>> dup binary <client> <connection> >>connection
+    keep-connection swap send ;
+
+: disconnect ( remote-thread -- )
+    dup connection>> [
+        [ terminate-connection over send ] dip stream>> dispose
+    ] when* f >>connection drop ;
+
+: with-connection ( remote-thread quot -- )
+    '[ connect @ ] over [ disconnect ] curry [ ] cleanup ; inline
 
 : send-remote-message ( message node -- )
     binary [ serialize ] with-client ;
 
+: with-stream* ( duplex-stream quot -- )
```

Comment:
I think this is duplicative of ``io.streams.duplex:with-stream*``...

Reply:
Wow, you are right, this is completely redundant. I'll remove it.